api/utils/auth.js

```diff
@@ -0,0 +1,23 @@
+const jwt = require("jsonwebtoken");
+
+const { userDao } = require("../models");
+
+const loginRequired = async (req, res, next) => {
```

Comment:
```suggestion
const loginRequired = catchAsync(async (req, res, next) => {
```
decoded라는 이름으로 작성된 jwt.verify() 코드는 외부 모듈을 활용하므로 if(!decoded)와 같이 핸들링 할 수 없습니다. 때문에 토큰에 조금의 변형이 일어난 상태로 들어오게 되면 지금의 코드에서는 서버가 처리하고 있지 않은 예외 상황이 발생하여 down 되게 됩니다. 따라서 suggestion처럼 catchAsync를 활용하거나 필요한 부분에 try catch를 활용해 주셔야겠습니다.